test/gtest/lstm_extra.cpp

```diff
@@ -0,0 +1,204 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "lstm.hpp"
+#include "get_handle.hpp"
+#include <miopen/env.hpp>
+#include <gtest/gtest_common.hpp>
+#include <gtest/gtest.h>
+#include <boost/algorithm/string.hpp>
+
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_FLOAT)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_HALF)
+
+namespace lstm_extra {
+using EnvType = std::tuple<std::pair<miopen::env::MIOPEN_TEST_ALL, std::string_view>,
+                           std::pair<miopen::env::MIOPEN_TEST_FLOAT, std::string_view>,
+                           std::pair<miopen::env::MIOPEN_TEST_HALF, std::string_view>>;
+
+bool Skip(miopenDataType_t prec)
+{
+    bool flag = !miopen::IsEnabled(ENV(MIOPEN_TEST_ALL));
+    switch(prec)
+    {
+    case miopenFloat: return flag && !miopen::IsEnabled(ENV(MIOPEN_TEST_FLOAT));
+    case miopenHalf: return flag && !miopen::IsEnabled(ENV(MIOPEN_TEST_HALF));
+    case miopenFloat8:
+    case miopenBFloat8:
+    case miopenInt8:
+    case miopenBFloat16:
+    case miopenInt32:
+    case miopenDouble:
+    default: MIOPEN_THROW("Unsupported datatype");
+    }
+    return true;
+}
+
+void GetArgs(const std::string& param, std::vector<std::string>& tokens)
+{
+    std::stringstream ss(param);
+    std::istream_iterator<std::string> begin(ss);
+    std::istream_iterator<std::string> end;
+    while(begin != end)
+        tokens.push_back(*begin++);
+}
+
+void SetEnv(EnvType env_vars)
+{
+    std::apply(
+        [](const auto&... pairs) {
+            (..., (miopen::UpdateEnvVar(pairs.first, std::string_view(pairs.second))));
+        },
+        env_vars);
+}
+
+auto GetTestCases(std::string precision)
+{
+    const auto env = std::tuple{
+        std::pair{ENV(MIOPEN_TEST_ALL), std::string_view("ON")},
+        std::pair{ENV(MIOPEN_TEST_FLOAT),
+                  precision == "--float" ? std::string_view("ON") : std::string_view("OFF")},
+        std::pair{ENV(MIOPEN_TEST_HALF),
+                  precision == "--half" ? std::string_view("ON") : std::string_view("OFF")}};
+
+    std::string flags       = "test_lstm --verbose " + precision;
```

Comment:
Always pass "--float" or nothing.

[Warning] Please check if options are parsed correctly by the test. I remember some quirks related to the ordering of options.

Reply:
[resolved]